app/assets/javascripts/controllers/middleware_providers/module_controllers.js

```diff
@@ -0,0 +1,144 @@
+miqHttpInject(angular.module('mwProviders.controllers', ['miQStaticAssets']))
+.controller('miqNewProviderController', ['$location', '$http', function($location, $http) {
+  this.onBackToListClick = _.bind(function() {
+    $location.path('/ems_middleware/show_list');
+  }, this);
+
+  this.validateAction = _.bind(function(validateData) {
+    $http.post('/ems_middleware/validate_provider', this.newProvider).then(function (response) {
+      console.log(response.data);
+    });
+  }, this);
+
+  this.formActions = [{
+    btnClass: 'btn-primary',
+    title: 'Add this Middleware Manager',
+    clickFunction: _.bind(function() {
+      console.log('You\'ve just added new Middleware provider', this.newProvider);
+    }, this),
+    label: 'Add'
+  }, {
+    btnClass: 'btn-default',
+    title: 'Cancel and return to list of Middleware providers',
+    clickFunction: _.bind(function() {
+      this.onBackToListClick();
+    }, this),
+    label: 'Cancel'
+  }];
+
+  this.credentialsTabs = [{
+    title: 'Default',
+    description: 'Required. Should have privileged access, such as root or administrator.',
+    modelKey: 'default'
+  }, {
+    title: 'Remote login',
+    description: 'Required if SSH login is disabled for the Default account.',
+    modelKey: 'remoteLogin'
+  }, {
+    title: 'Web services',
+    description: 'Used for access to Web Services.',
+    modelKey: 'webServices'
+  }, {
+    title: 'IPMI',
+    description: 'Used for access to IPMI.',
+    modelKey: 'impi'
+  }];
+
+  this.types = ['Hawkular'];
+
+  this.zones = ['Dan', 'Dan 2 Zone', 'Default Zone', 'RHEV 1 for fleecing'];
+
+  this.newProvider = {
+    type: 'default',
+    zone: 'Default Zone',
+    remoteLogin: {},
+    webServices: {},
+    impi: {}
+  };
+}])
+.controller('miqListProvidersController', ['MiQDataTableService', '$location', function(MiQDataTableService, $location) {
+    this.data = [];
+    this.columnsToShow = [];
+    this.defaultAction = {
+      id: 'new_provider',
+      title: 'Add a New Middleware Provider',
+      icon: 'pficon pficon-add-circle-o fa-lg',
+      actionFunction: _.bind(function() {
+        $location.path('/ems_middleware/new');
+      }, this)
+    };
+    MiQDataTableService.retrieveRowsAndColumnsFromUrl(
+      '/ems_middleware/list_providers'
+    ).then((rowsCols) => {
```

Comment:
I kept some arrow functions there since I wanted to make it faster so you can look at it, I will rewrite all these arrow functions to `_.bind(function(){}, this)`.